pkg/sql/catalog/validate.go

```diff
@@ -395,7 +396,15 @@ func (vdg *validationDescGetterImpl) GetTypeDescriptor(id descpb.ID) (TypeDescri
 	if !found || desc == nil {
 		return nil, WrapTypeDescRefErr(id, ErrDescriptorNotFound)
 	}
-	return AsTypeDescriptor(desc)
+	if _, ok := desc.(TableDescriptor); ok {
+		return nil, unimplemented.NewWithIssuef(70099, "table record type %q not allowed in definition",
+			desc.GetName())
```

Comment:
This makes me uneasy. We should not be assuming that the descriptor holding the reference intended that ID to be to a record type descriptor. It could be the result of just about any kind of descriptor corruption, we don't know.

Unless this serves a purpose I'm not clued in to, I'd prefer to remove this if-block entirely.